ocf_datapipes/load/pv.py

```diff
@@ -0,0 +1,243 @@
+import datetime
+import logging
+from pathlib import Path
+from typing import Optional, Union
+
+import fsspec
+import numpy as np
+import pandas as pd
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+_log = logging.getLogger(__name__)
+
+
+@functional_datapipe("open_pv_netcdf")
+class OpenPVFromNetCDFIterDataPipe(IterDataPipe):
+    def __init__(
+        self,
+        pv_power_filename: str,
+        pv_metadata_filename: str,
+        n_pv_systems_per_example: int,
+        sample_period_duration: datetime.timedelta = datetime.timedelta(minutes=5),
+    ):
+        super().__init__()
+        self.pv_power_filename = pv_power_filename
+        self.pv_metadata_filename = pv_metadata_filename
+        self.n_pv_systems_per_example = n_pv_systems_per_example
+        self.sample_period_duration = sample_period_duration
+
+    def __iter__(self):
+        data: xr.DataArray = load_everything_into_ram(
+            self.pv_power_filename, self.pv_metadata_filename, self.sample_period_duration
+        )
+        while True:
+            yield data
+
+
+@functional_datapipe("open_pv_from_db")
+class OpenPVFromDBIterDataPipe(IterDataPipe):
+    def __init__(self):
+        super().__init__()
+
+    def __iter__(self):
+        pass
+
+
+def load_everything_into_ram(
+    pv_power_filename, pv_metadata_filename, sample_period_duration
+) -> xr.DataArray:
+    """Open AND load PV data into RAM."""
+    # Load pd.DataFrame of power and pd.Series of capacities:
+    pv_power_watts, pv_capacity_wp, pv_system_row_number = _load_pv_power_watts_and_capacity_wp(
+        pv_power_filename,
+    )
+    pv_metadata = _load_pv_metadata(pv_metadata_filename)
+    # Ensure pv_metadata, pv_power_watts, and pv_capacity_wp all have the same set of
+    # PV system IDs, in the same order:
+    pv_metadata, pv_power_watts = _intersection_of_pv_system_ids(pv_metadata, pv_power_watts)
```

Comment:
Would be good if all these was in a seprate function, then the live db code could also use it